src/components/popovers/Popover.jsx

```diff
@@ -69,189 +69,196 @@ const useCloseOnScroll = (closeOnScrollRef, isActive, closeCallback) => {
  * @param {Object} props react props
  * @returns {ReactElement}
  */
-const Popover = ({
-  interactionMode = 'click',
-  disablePortal,
-  position = 'auto',
-  alignment = 'start',
-  distance = 4,
-  delay = 0,
-  onOpen = () => {},
-  onClose = () => {},
-  trigger,
-  children,
-  isOpen,
-  transitionName,
-  disableScrollRef,
-  closeOnScrollRef,
-}) => {
-  const [isActive, setIsActive] = useState(false);
-  const refTriggerWrap = useRef(null);
-  const refContent = useRef(null);
-
-  const handleSetIsActive = (newValue) => {
-    setIsActive(() => {
-      if (newValue) {
-        onOpen();
-      } else {
-        onClose();
-      }
-      return newValue;
-    });
-  };
+const Popover = React.forwardRef(
+  (
+    {
+      interactionMode = 'click',
+      disablePortal,
+      position = 'auto',
+      alignment = 'start',
+      distance = 4,
+      delay = 0,
+      onOpen = () => {},
+      onClose = () => {},
+      trigger,
+      children,
+      isOpen,
+      transitionName,
+      disableScrollRef,
+      closeOnScrollRef,
+    },
+    forwardedRef
+  ) => {
+    const [isActive, setIsActive] = useState(false);
+    const refTriggerWrap = useRef(null);
+    const refContent = forwardedRef || useRef(null);
```

Comment:
this is the main thing that changed - if a `ref` is passed, we use the `forwardRef` instead of creating a new ref to use internally.

Reply:
I think it's against hooks rules to define them conditionally. Could the conditional be defined in a different place?

<img width="619" alt="Screen Shot 2020-06-03 at 3 34 50 PM" src="https://user-images.githubusercontent.com/708820/83681031-cb7c4400-a5af-11ea-85a4-92e7302e89cc.png">

Reply:
i think i "fixed" this by using the `createRef` api for this ref instead